File: src/main/java/co/electric/snake/model/strategy/DefaultStrategy.java
package co.electric.snake.model.strategy;

import co.electric.snake.model.Arena;
import co.electric.snake.model.Coordinate;
import co.electric.snake.model.Direction;
import co.electric.snake.model.Snake;

public class DefaultStrategy implements SnakeStrategy {

    @Override
    public Direction nextMove(Snake snake, Arena arena) {
        final Coordinate startCoordinate = snake.getHeadCoordinate();
        final Coordinate foodCoordinate = arena.getFood().get(0).getCoordinate();

        int minDistance = Integer.MAX_VALUE;
        Direction bestDirection = Direction.WEST;

        for (Direction direction : Direction.values()) {
            Coordinate nextCoordinate = arena.nextCoordinate(startCoordinate, direction);

            int actualDistance = nextCoordinate.minDistance(foodCoordinate, arena.getMaxCoordinate());

            if (minDistance > actualDistance && !arena.isOccupied(nextCoordinate)) {
                minDistance = actualDistance;
                bestDirection = direction;
            }
        }

        return bestDirection;
    }

}

File: src/main/kotlin/co/electric/snake/model/strategy/DefaultStrategy.kt
package co.electric.snake.model.strategy

import co.electric.snake.model.Arena
import co.electric.snake.model.Direction
import co.electric.snake.model.Snake
import java.util.*

class DefaultStrategy : SnakeStrategy {

    override fun nextMove(snake: Snake, arena: Arena): Direction {
        val startCoordinate = snake.getHeadCoordinate()
        val foodCoordinate = arena.food[0].coordinate
        var minDistance = Int.MAX_VALUE
        var bestDirection: Direction = Direction.WEST
        Arrays.stream(Direction.values()).forEach { direction ->
            val nextCoordinate = arena.nextCoordinate(startCoordinate, direction)
            val actualDistance = nextCoordinate.minDistance(foodCoordinate, arena.maxCoordinate)
            if (minDistance > actualDistance && !arena.isOccupied(nextCoordinate)) {
                minDistance = actualDistance
                bestDirection = direction
            }
        }
        return bestDirection
    }

}
